### PR TITLE
Create User menu

### PR DESCRIPTION
**Pull request by miguelrodrigues22**
Create User Menu with

- View Profile
- My Photos
- Settings
- Sign Out

Create empty pages for

- View Profile
- Settings

Create My Photos page like the Collections page

### CODE REVIEW

**Review by tiagocardoso98 — APPROVED** [review]

File: components/helpers/interfaces.tsx [hunk 1]
```diff
@@ -32,10 +33,25 @@ export interface Collection {
   picture: Picture[]
   user: string
 }
+
+export interface MyPhoto {
+  id: number
+  name: string
+  picture: Picture[]
```
Comment (JdFSilva) [inline components/helpers/interfaces.tsx:40]: should this be named `pictures` instead?

File: components/helpers/interfaces.tsx [hunk 2]
```diff
@@ -32,10 +33,25 @@ export interface Collection {
   picture: Picture[]
   user: string
 }
+
+export interface MyPhoto {
```
Comment (JdFSilva) [inline components/helpers/interfaces.tsx:37]: should this be named `MyPhotos` instead?

File: components/helpers/interfaces.tsx [hunk 3]
```diff
@@ -32,10 +33,25 @@ export interface Collection {
...
 
+export interface MyPhotosProps {
+  myphotosList: MyPhoto[]
+}
+
+export interface MyPhotoProps {
+  myphoto: MyPhoto
+}
+
 export interface ContestFilterProps {
```
Comment (JdFSilva) [inline components/helpers/interfaces.tsx:55]: keep the props on the component files

File: components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+                      alt='profile'
+                      width={40}
+                      height={40}
+                      className='mr-2'
+                    />
+                    <span>Sign Out</span>
+                  </button>
+                </ul>
+              </div>
+            )}
```
Comment (JdFSilva) [inline components/Navbar/Navbar.tsx:236]: this should be on it's own component

File: pages/myphotos/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { MyPhotosProps } from '../../components/helpers/interfaces'
+import MyPhoto from '../../components/MyPhoto'
+import { getMyPhotosList } from '../../services/reventService'
+
+export default function MyPhotos({ myphotosList }: MyPhotosProps) {
```
Comment (JdFSilva) [inline pages/myphotos/index.tsx:5]: move the props into this file

File: components/MyPhoto.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import { MyPhotoProps } from './helpers/interfaces'
+
+export default function MyPhoto({ myphoto }: MyPhotoProps) {
+  return (
+    <>
+      <div className='max-w-sm rounded overflow-hidden shadow-lg'>
+        <img className='w-full' src='/images/contest.jpeg' alt='' />
+
+        <div className='px-6 py-4 text-center'>
+          <div className='font-bold text-xl mb-2'>{myphoto.name}</div>
```
[the remark this answers is not in the document]
Comment (miguelrodrigues22) [reply components/MyPhoto.tsx:11]: it had in my static service yes, will change it to description

File: components/Navbar/NavigationMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import { useRouter } from 'next/router'
+import React from 'react'
+import Image from 'next/image'
+
+interface setShowUserMenu {
+  setShowUserMenu: React.Dispatch<React.SetStateAction<boolean>>
+}
+export default function NavigationMenu({ setShowUserMenu }: setShowUserMenu) {
```
Comment (JdFSilva) [inline components/Navbar/NavigationMenu.tsx:8]: can we rename `NavigationMenu` to `UserMenu`?

File: components/Navbar/NavigationMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import { useRouter } from 'next/router'
+import React from 'react'
+import Image from 'next/image'
+
+interface setShowUserMenu {
```
Comment (JdFSilva) [inline components/Navbar/NavigationMenu.tsx:5]: name of the props should be `NameOfTheComponentProps`

File: components/Navbar/Navbar.tsx [hunk 9]
```diff
@@ -176,63 +177,7 @@ export default function Navbar() {
...
-                      alt='profile'
-                      width={40}
-                      height={40}
-                      className='mr-2'
-                    />
-                    <span>Sign Out</span>
-                  </button>
-                </ul>
-              </div>
+              <NavigationMenu setShowUserMenu={setShowUserMenu} />
```
Comment (JdFSilva) [inline components/Navbar/Navbar.tsx:180]: UserMenu?

**Review by JdFSilva — CHANGES_REQUESTED** [review]

File: components/Navbar/Navbar.tsx (hunk 9 above)
Comment (miguelrodrigues22) [reply components/Navbar/Navbar.tsx:180]: what about it?

**Review by JdFSilva — APPROVED** [review]

**Pull request merged**